### DISCUSSION TITLE
Localization in language server

### DISCUSSION THREAD

**Discussion by WideSu** [eclipse-langium/langium#1658] [Q&A]
Hi langium community, I am trying to make the documentation, details and tooltip localized to different countries according to users' language setting. Do you know how to localize the texts in the language server?

**Comment by msujew** [comment]
Hey @WideSu,

we don't provide anything for this ourselves. I would recommend you to use the [`@vscode/l10n`](https://github.com/microsoft/vscode-l10n/tree/main/l10n) library which is specificially made for this use case.

See also [this language server entry point](https://github.com/Microsoft/vscode/blob/main/extensions/css-language-features/server/src/node/cssServerNodeMain.ts) to see how to set up the localization configuration.
